Tests/MongoSwiftTests/MongoDatabaseTests.swift

```diff
@@ -42,6 +42,47 @@ final class MongoDatabaseTests: MongoSwiftTestCase {
                 .to(throwError(ServerError.commandError(code: 59, message: "", errorLabels: nil)))
     }
 
+    func testDropDatabase() throws {
+        let center = NotificationCenter.default
+
+        let client = try MongoClient(options: ClientOptions(eventMonitoring: true))
+        client.enableMonitoring(forEvents: .commandMonitoring)
+
+        var db = client.db(type(of: self).testDatabase)
+        var writeConcern = try WriteConcern(journal: true, w: .number(1))
+
+        let collection = db.collection("collection")
+        try collection.insertOne(["test": "blahblah"])
+
+        var expectedWriteConcerns: [Document] = [
+                                                    ["w": Int32(1), "j": true],
```

Comment:
To make this test a little less brittle, I think it would be better to define the array of expected write concerns as an array of the actual type (e.g. `[WriteConcern]`). Then, in the event handler we can encode the popped one and compare (e.g. `encoder.encode(expectedWriteConcerns.removeFirst())`). This removes the dependency on the encoded format of `WriteConcern` from this test, so that when the C driver fixes the bug requiring int32's for wtimeoutms, this test won't have to be updated. 

With this change, below when passing in the write concerns, we can index on `expectedWriteConcerns` itself instead of constructing write concerns that correspond to its entries.